# Marketing Campaign Analysis

## Business Understanding
### Overview
Podha, a growing company, has been running multiple marketing campaigns to acquire new customers. However, due to a crunch in funds, the company has decided to limit its efforts to a single marketing campaign that optimizes budget utilization while maximizing customer acquisition and profitability. Additionally, there is a need to predict customer acquisition based on the available budget.

### Business Problem
The company must identify which marketing campaign has brought in the most profitable customers to ensure financial stability and growth. Furthermore, predicting customer acquisition based on budget will help in strategic planning and resource allocation.

Therefore, I aim to:
1. Evaluate the performance of past marketing campaigns.
2. Determine the best acquisition source based on profitability metrics.
3. Predict customer acquisition based on Podha's budget.
4. Provide actionable recommendations supported by data visualization to inform strategic decisions.


## Data Understanding
Here, I intend to comprehensively explore and analyze our dataset to gain insights into its structure, content, and quality.

In [1]:
# Import necessary libraries

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Load the dataset
file_path = 'data/Order_Data_meriskill.csv'
df = pd.read_csv(file_path)

c:\Users\hp\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (3,10) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [3]:
# View the first five rows of the dataset to see if loading has been done correctly
df.head()

OrderID          Region         Country CustID    Customer_Name  \
0  SO45080       Northwest   United States  14657      JOHN THOMAS   
1  SO45079       Southwest   United States  29255  KYLE WASHINGTON   
2  SO45082       Australia       Australia  11455        ROSS SANZ   
3  SO45081          Canada          Canada  26782       SETH LEWIS   
4  SO45083  United Kingdom  United Kingdom  14947   ALEJANDRO CHEN   

   ProductSKU Product_Category  OrderLineItem  OrderQuantity  ProductCost  \
0  BK-R50B-58           Plants              1            1.0     413.1463   
1  BK-R93R-48           Plants              1            1.0    2171.2942   
2  BK-M82B-44           Plants              1            1.0    1898.0944   
3  BK-R50B-44           Plants              1            1.0     413.1463   
4  BK-R93R-48           Plants              1            1.0    2171.2942   

  ProductPrice   OrderDate AcquisitionSource      TransactionID  Fraud  \
0     699.0982  01-01-2020        Meta-ads\r  I4127227587819694  False   
1      3578.27  01-01-2020      Google-ads\r  E9525730529761877  False   
2      3374.99  01-01-2020      Google-ads\r  Z6376643485803420  False   
3     699.0982  01-01-2020      Google-ads\r  R8609290025411138  False   
4      3578.27  02-01-2020        Meta-ads\r  Y9572700698484288  False   

  PaymentMethod  CardType Gender  
0    CREDITCARD  DISCOVER      M  
1    CREDITCARD        MC      M  
2    CREDITCARD      VISA      M  
3    CREDITCARD      VISA      M  
4    CREDITCARD      VISA      M

In [4]:
# View the overallinformation of each column
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55910 entries, 0 to 55909
Data columns (total 18 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   OrderID            55910 non-null  object 
 1   Region             55910 non-null  object 
 2   Country            55910 non-null  object 
 3   CustID             55910 non-null  object 
 4   Customer_Name      55907 non-null  object 
 5   ProductSKU         55910 non-null  object 
 6   Product_Category   55910 non-null  object 
 7   OrderLineItem      55910 non-null  int64  
 8   OrderQuantity      55910 non-null  float64
 9   ProductCost        55910 non-null  float64
 10  ProductPrice       55910 non-null  object 
 11  OrderDate          55910 non-null  object 
 12  AcquisitionSource  55909 non-null  object 
 13  TransactionID      55910 non-null  object 
 14  Fraud              55207 non-null  object 
 15  PaymentMethod      55910 non-null  object 
 16  CardType           506

In [5]:
# Summary statistics of numerical columns
df.describe()

OrderLineItem  OrderQuantity   ProductCost
count   55910.000000   55910.000000  55910.000000
mean        1.903237       1.501607    256.143102
std         1.021286       0.612305    496.563101
min         1.000000       1.000000      0.856500
25%         1.000000       1.000000      2.973300
50%         2.000000       1.000000     11.216300
75%         2.000000       2.000000     59.466000
max         8.000000       3.000000   2171.294200

### Data Overview
The dataset provides comprehensive information about customer transactions, including their demographics, purchasing behavior, product preferences, and the sources from which they were acquired.
- **Rows**: 55,910.
- **Columns**: 18.
- **Column Names**: 
     - `Customer Identification and Demographics`: CustID, Customer_Name, Gender.
     - `Transaction and Order Details`: OrderID, OrderLineItem, OrderQuantity, OrderDate, TransactionID.
     - `Geographic Data`: Region, Country.
     - `Product Information`: ProductSKU, Product_Category, ProductCost, ProductPrice.
     - `Customer Acquisition Data`: AcquisitionSource.
     - `Payment Information`: PaymentMethod, CardType.
     - `Risk and Fraud`: Fraud.
- **Data Types**:
     - `Numerical Columns`: 3.
     - `Categorical Columns`: 15.

     **NOTE**: `ProductPrice` is stored as an object and needs to be converted to a numeric format for meaningful statistics.
- **Missing Values**:
     - `Customer_Name`: 3 missing values.
     - `AcquisitionSource`: 1 missing value.
     - `Fraud`: 703 missing values.
     - `CardType`: 5,223 missing values.
     - `Gender`: 6,328 missing values.
- **Relevance to Objectives**
     - `Marketing Campaign Analysis`: Columns such as AcquisitionSource, ProductCost, ProductPrice, and OrderQuantity are directly relevant to calculating profitability and identifying effective campaigns.
     - `Customer Acquisition Prediction`: Fields like Region, Country, AcquisitionSource, and budget-related fields (ProductCost, ProductPrice) can be used to build predictive models.